src/main/java/software/amazon/nio/spi/s3/S3TransferException.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.nio.spi.s3;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+
+public class S3TransferException extends IOException {
```

Comment:
It would be valuable to have javadocs to explain what this exception means and when it should be used. Especially as it is a `public` class

Reply:
Agreed, I added some JavaDoc to explain the purpose and how it is meant to be used.